pkg/server/test/write_authzmodel.go

```diff
@@ -428,6 +433,166 @@ func WriteAuthorizationModelTest(t *testing.T, datastore storage.OpenFGADatastor
 				fmt.Errorf("type 'user' defines a relation with an empty string for a name"),
 			),
 		},
+
+		// conditions
+		{
+			name: "condition_fails_undefined",
```

Comment:
can you add a test that asserts that you can't have more than one condition with the same name?

Reply:
there's one that asserts the case for `condition key '%s' does not match condition name '%s'`. because of this same guard, you can't really have conditions with the same name, since that'd imply a map with duplicate keys.

Reply:
Another +1 to using maps 😉